### PR TITLE
Add a manager for timed events and display event banners during events

### PR DESCRIPTION
**Pull request by IceFire03**
Currently, weekend events are handled in a hacky manner, just adding a date check and shiny multiplier into pokemon.ts; this PR implements a TimedEventManager to handle events in a more standardized and safe manner. New events can be entered via the TimedEvent interface and can be scheduled to occur during certain dates. Then, banners can easily be added to the events folder and will be displayed on the title screen during events. Tested for all screen sizes and for events and edge cases.
Screenshot 2024-05-17 at 7 01 23 PM

### CODE REVIEW

**Review by bennybroseph — DISMISSED** [review]

**Comment by bennybroseph** [comment]
I like this as is where it's already placed. Timed events are hard to test, but assuming this works as intended it's a yes from me.

File: src/loading-scene.ts [hunk 1]
```diff
@@ -209,9 +213,6 @@ export class LoadingScene extends SceneBase {
 
     this.loadImage('egg_list_bg', 'ui');
 
-    this.loadImage('end_m', 'cg');
-    this.loadImage('end_f', 'cg');
-
```
Comment (TempsRay) [inline src/loading-scene.ts:214]: Is this intentional?

Comment (IceFire03) [reply src/loading-scene.ts]: No, I think my branch wasn't properly synced. Fixed!

**Comment by IceFire03** [comment]
> I like this as is where it's already placed. Timed events are hard to test, but assuming this works as intended it's a yes from me.

Agreed, I tested the timing extensively by actually changing the date/time on the local server and it worked without a hitch.

**Comment by brain-frog** [comment]
We all of a sudden last minute needed to add this functionality so I very much appreciate the fact that you made this. I implemented it based on this with some adjustments.

**Pull request closed without merging**